garrysmod/gamemodes/base/entities/entities/lua_run.lua

```diff
@@ -38,7 +38,7 @@ function ENT:RunCode( activator, caller, code )
 
 	self:SetupGlobals( activator, caller )
 
-		RunString( code )
+		RunString( code, game.GetMap() )

```

Comment:
As concatenation isn't expensive, and there aren't many run_lua entities per map, in general, it probably won't impact performance in any meaningful way, so it doesn't matter when it's done really.